consensus/misc/eip1559.go

```diff
@@ -39,7 +39,7 @@ func VerifyEip1559Header(config *params.ChainConfig, parent, header *types.Heade
 		return err
 	}
 	// Verify the header is not malformed
-	if header.BaseFee == nil && config.Scroll.BaseFeeEnabled() {
+	if (header.BaseFee == nil && config.Scroll.BaseFeeEnabled()) || (header.BaseFee != nil && !config.Scroll.BaseFeeEnabled()) {
```

Comment:
Need a different error message for the new case

Reply:
updated